### ISSUE TITLE
Introduce something more about Codegen and LLVM

### ISSUE THREAD

**Issue by crlf0710** [rust-lang/rustc-guide#442]
Hello i'm trying to make a little progress on the raw_dylib rfc implementation. It seems this RFC needs to deal with the codegen process, and i'm seeking for more introductory materials on this affair. It will be great if this can be introduced more in rustc-guide. Thanks!

**Comment by mark-i-m** [comment]
@crlf0710 Yes, this is one of the biggest gaps in the guide that I've been wanting to fill for a while.

Would you be willing to write down what you learned?

@spastorino perhaps we can schedule a zulip discussion with nagisa

**Comment by crlf0710** [comment]
Sure, I'm happy to help with the guide as much as i can.

**Comment by bjorn3** [comment]
I wrote some advice from my experience writing a codegen backend at https://github.com/bjorn3/rustc_codegen_cranelift/issues/647. Maybe include some of it in the guide?
